lib/client/native_client.dart

```diff
@@ -0,0 +1,178 @@
+import 'dart:ffi';
+import 'dart:io' show Platform;
+
+import 'package:ffi/ffi.dart';
+
+typedef InitDartApiNative = IntPtr Function(Pointer<Void>);
+typedef InitDartApi = int Function(Pointer<Void>);
+
+typedef NewClientNative = IntPtr Function(
+    Pointer<Utf8> appId,
+    Pointer<Utf8> rendezvousUrl,
+    Pointer<Utf8> transitRelayUrl,
+    Int32 passPhraseComponentLength);
+typedef NewClient = int Function(
+    Pointer<Utf8> appId,
+    Pointer<Utf8> rendezvousUrl,
+    Pointer<Utf8> transitRelayUrl,
+    int passPhraseComponentLength);
+
+typedef ClientSendTextNative = Int32 Function(Uint32 goClientId,
+    Pointer<Utf8> msg, Pointer<Pointer<Utf8>> codeOut, Int32 callbackPortId);
+
+typedef ClientSendText = int Function(int goClientId, Pointer<Utf8> msg,
+    Pointer<Pointer<Utf8>> codeOut, int callbackPortId);
+
+typedef ClientSendFileNative = Int32 Function(
+    Uint32 goClientId,
+    Pointer<Utf8> fileName,
+    Uint32 length,
+    Pointer<Uint8> fileBytes,
+    Pointer<Pointer<Utf8>> codeOut,
+    Int32 callbackPortId);
+
+typedef ClientSendFile = int Function(
+    int goClientId,
+    Pointer<Utf8> fileName,
+    int length,
+    Pointer<Uint8> fileBytes,
+    Pointer<Pointer<Utf8>> codePtr,
+    int callbackPortId);
+
+typedef ClientRecvTextNative = Int32 Function(
+    Uint32 goClientId, Pointer<Utf8> code, Int32 callbackPortId);
+
+typedef ClientRecvText = int Function(
+    int goClientId, Pointer<Utf8> code, int callbackPortId);
+
+typedef ClientRecvFileNative = Int32 Function(
+    Uint32 goClientId, Pointer<Utf8> code, Int32 callbackPortId);
+
+typedef ClientRecvFile = int Function(
+    int goClientId, Pointer<Utf8> code, int callbackPortId);
+
+const DEFAULT_APP_ID = "lothar.com/wormhole/text-or-file-xfer";
+const DEFAULT_RENDEZVOUS_URL = "ws://relay.magic-wormhole.io:4000/v1";
+const DEFAULT_TRANSIT_RELAY_URL = "tcp:transit.magic-wormhole.io:4001";
+const DEFAULT_PASSPHRASE_COMPONENT_LENGTH = 2;
+
+class Config {
+  final String appId;
+  final String rendezvousUrl;
+  final String transitRelayUrl;
+  final int passPhraseComponentLength;
+
+  Config(
+      {this.appId = DEFAULT_APP_ID,
+      this.rendezvousUrl = DEFAULT_RENDEZVOUS_URL,
+      this.transitRelayUrl = DEFAULT_TRANSIT_RELAY_URL,
+      this.passPhraseComponentLength = DEFAULT_PASSPHRASE_COMPONENT_LENGTH});
+}
+
+class NativeClient {
+  late final DynamicLibrary _wormholeWilliamLib;
+  late final DynamicLibrary _asyncCallbackLib;
+  late final _goClientId;
+
+  String _appId = DEFAULT_APP_ID;
+  String _rendezvousUrl = DEFAULT_RENDEZVOUS_URL;
+  String _transitRelayUrl = DEFAULT_TRANSIT_RELAY_URL;
+  int _passPhraseComponentLength = DEFAULT_PASSPHRASE_COMPONENT_LENGTH;
```

Comment:
Simplify. We can use a new / default config in the constructor.

Reply:
Did a bit of refactoring to clean this up. `NativeClient` now either uses the provided `Config` or creates a `Config` with no explicit parameters, thereby using the defaults.